Integrations/integration-CounterTack.yml

```diff
@@ -0,0 +1,1716 @@
+    # Headers to be sent in requests
+    GET_HEADERS = {
+        'Content-Type': 'application/json'
+    }
+
+
+    def http_request(method, url, headers={}, body=None, body_params=None):
+        """
+        returns the http request
+
+        """
+        #basic authentication
+        try:
+            response = requests.request(
+                method,
+                url,
+                auth=(USERNAME,PASSWORD),
+                headers=headers,
+                verify=USE_SSL,
+                data=body
+        )
+        except requests.exceptions.SSLError as e:
+            LOG(e.message)
+            raise ValueError('An SSL error occurred when trying to connect to the server. Consider configuring unsecure connection in the integration settings.')
+
+        #handle request failure
+        if response.status_code not in range(200,205):
+            message = parse_error_response(response)
+            raise ValueError('Request failed with status code {}\n{}'.format(response.status_code, message))
+
+        return response
+
+
+    def parse_error_response(response):
+        try:
+            res = response.json()
+            msg = res.get('message')
+            if res.get('details') is not None and res.get('details')[0].get('message') is not None:
+                msg = msg + "\n"+json.dumps(res.get('details')[0])
+        except Exception as e:
+            return response.text
+        return msg
+
+    def return_error_entry(message):
+        error_entry = {
+            'Type': entryTypes['error'],
+            'Contents': message,
+            'ContentsFormat': formats['text']
+        }
+        demisto.results(error_entry)
+
+    def behavior_to_incident(behavior):
+            incident = {}
+            incident['name'] = 'CounterTack Behavior Triggered: ' + behavior['name']
+            incident['occurred'] = behavior['last_found'] + 'Z'
+            incident['rawJSON'] = json.dumps(behavior)
+
+            return incident
+
+    """
+
+    ENDPOINTS
+
+    """
+
+    def get_endpoints_request():
+        """
+        This request returns a collection of endpoints.
+        """
+
+        url = '{}/endpoints'.format(BASE_PATH)
+        response = http_request('GET',url,headers=GET_HEADERS)
+        return response.json()
+
+    def get_endpoints():
+
+        """
+        Returns the information on existing endpoints
+        """
+        headers = ['OS','Name','Threat','Status','ID','IP']
+        data = []
+
+        endpoints = get_endpoints_request()
+        for endpoint in endpoints:
+            data.append ({
+                'ID': endpoint.get('id'),
+                'Name': endpoint.get('name'),
+                'OS': endpoint.get('product_name'),
+                'IP': endpoint.get('ips'),
+                'Status': endpoint.get('status'),
+                'Threat': endpoint.get('threat')
+            })
+
+        context['CounterTack.Endpoint(val.ID && val.ID === obj.ID)'] = endpoints
```

Comment:
`endpoints` have no key `ID`, but `id`
need to process the data for the context like for the human readable